pkg/depend/request.go

```diff
@@ -1,37 +1,89 @@
 package depend
 
-import "encoding/json"
+import (
+	"encoding/json"
+	"errors"
 
-// Repository holds all the information about the merge which is taking place.
-// Dependencies will only contain the URL and Branch.
+	"github.com/git-depend/git-depend/pkg/git"
+)
+
+// Request holds all the information about the merge which is taking place.
 type Request struct {
-	Url    string     `json:"Url"`
-	Branch string     `json:"Branch"`
-	Author string     `json:"Author,omitempty"`
-	Email  string     `json:"Email,omitempty"`
-	Date   string     `json:"Date,omitempty"`
-	Deps   *[]Request `json:"Deps,omitempty"`
-}
-
-// NewRequest creates the Request struct from the given fields
-func NewRequest(url string, branch string, author string, email string, date string, deps *[]Request) *Request {
-	return &Request{
-		url,
-		branch,
-		author,
-		email,
-		date,
-		deps,
+	Name   string `json:"Name"`
+	From   string `json:"From"`
+	To     string `json:"To"`
+	Author string `json:"Author,omitempty"`
+	Email  string `json:"Email,omitempty"`
+}
+
+// Requests contains a map of Node names to the Request.
+type Requests struct {
+	Table     map[string]*Request
```

Comment:
Would it be cleaner to have `map[*Node]*Request` as the `Table` type? I'm also wondering if it would make it easier to follow if the naming of this were more explicit e.g. `NodeRequestMap` ?

Reply:
You're totally right, can't believe I missed that. Rewritten and greatly simplified!